grape/vertex_map/global_vertex_map.h

```diff
@@ -70,30 +70,19 @@ class GlobalVertexMapBuilder {
     const CommSpec& comm_spec = vertex_map.GetCommSpec();
     int worker_id = comm_spec.worker_id();
     int worker_num = comm_spec.worker_num();
-    fid_t fnum = comm_spec.fnum();
     {
       std::thread recv_thread([&]() {
         int src_worker_id = (worker_id + 1) % worker_num;
         while (src_worker_id != worker_id) {
-          for (fid_t fid = 0; fid < fnum; ++fid) {
-            if (comm_spec.FragToWorker(fid) != src_worker_id) {
-              continue;
-            }
-            sync_comm::Recv(vertex_map.indexers_[fid], src_worker_id, 0,
-                            comm_spec.comm());
-          }
+          sync_comm::Recv(vertex_map.indexers_[src_worker_id], src_worker_id, 0,
+                          comm_spec.comm());
           src_worker_id = (src_worker_id + 1) % worker_num;
         }
       });
       std::thread send_thread([&]() {
         int dst_worker_id = (worker_id + worker_num - 1) % worker_num;
         while (dst_worker_id != worker_id) {
-          for (fid_t fid = 0; fid < fnum; ++fid) {
-            if (comm_spec.FragToWorker(fid) != worker_id) {
-              continue;
-            }
-            sync_comm::Send(indexer_, dst_worker_id, 0, comm_spec.comm());
-          }
+          sync_comm::Send(indexer_, dst_worker_id, 0, comm_spec.comm());
           dst_worker_id = (dst_worker_id + worker_num - 1) % worker_num;
```

Comment:
`comm_spec.FragToWorker(fid) == worker_id` may not always holds if one worker has more than one fragments (it is not the case in libgrape-lite itself but might be implemented in the future).

So please keep this part as it was.

Reply:
done